libs/amount/amount.cpp

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <utility>
+
+#include "amount/amount.hpp"
+
+namespace amount {
+
+  using namespace boost::multiprecision;
+
+  Amount::Amount() : value_(0), precision_(0) {}
+
+  Amount::Amount(uint256_t value)
+      : value_(value), precision_(0) {}
+
+  Amount::Amount(uint256_t amount, uint8_t precision)
+      : value_(amount), precision_(precision) {}
+
+  Amount::Amount(const Amount &am)
+      : value_(am.value_), precision_(am.precision_) {}
+
+  Amount &Amount::operator=(const Amount &other) {
+    // check for self-assignment
+    if (&other == this) return *this;
+    std::copy(&other.value_, &other.value_ + sizeof(decltype(value_)), &value_);
+    std::copy(&other.precision_, &other.precision_ + sizeof(precision_),
+              &precision_);
+    return *this;
+  }
+
+  Amount::Amount(Amount &&am) : value_(0), precision_(0) {
+    value_ = am.value_;
+    precision_ = am.precision_;
+
+    am.value_ = 0;
+    am.precision_ = 0;
```

Comment:
A bit useless, standard specify for usage moved value UB, so don't need to reset this values